assets/src/scripts/monitoring-location/components/hydrograph/index.js

```diff
@@ -64,38 +64,41 @@ export const attachToNode = function(store,
         return;
     }
 
-    showDataLoadingIndicator(true);
+    showDataIndicators(true);
 
     const initialPeriod = startDT && endDT ? 'custom' : period || 'P7D';
     const initialStartTime = startDT ?
         DateTime.fromISO(startDT, {zone: config.locationTimeZone}).toISO() : null;
     const initialEndTime = endDT ?
         DateTime.fromISO(endDT, {zone: config.locationTimeZone}).endOf('day').toISO() : null;
+    const initialLoadCompare = compare === 'true' || compare === true ? true : false;
+    const thisShowOnlyGraph = showOnlyGraph === 'true' || showOnlyGraph === true ? true : false;
+    const thisShowMLName = showMLName === 'true' || showMLName === true ? true : false;
     const fetchHydrographDataPromise = store.dispatch(retrieveHydrographData(siteno, {
         parameterCode: parameterCode,
         period: initialPeriod === 'custom' ? null : initialPeriod,
         startTime: initialStartTime,
         endTime: initialEndTime,
-        loadCompare: compare,
+        loadCompare: initialLoadCompare,
         loadMedian: false
     }));
 
     // if showing the controls, fetch the parameters
     let fetchParametersPromise;
-    if (!showOnlyGraph) {
+    if (!thisShowOnlyGraph) {
         fetchParametersPromise = store.dispatch(retrieveHydrographParameters(siteno));
+    }
 
-        // Initialize all hydrograph state variables if showing the control
-        store.dispatch(setSelectedParameterCode(parameterCode));
-        store.dispatch(setCompareDataVisibility(compare));
-        if (period) {
-            store.dispatch(setSelectedDateRange(period));
-        } else if (startDT && endDT) {
-            store.dispatch(setSelectedDateRange('custom'));
-            store.dispatch(setSelectedCustomDateRange(startDT, endDT));
-        } else {
-            store.dispatch(setSelectedDateRange('P7D'));
-        }
+    // Initialize all hydrograph state variables if showing the control
+    store.dispatch(setSelectedParameterCode(parameterCode));
```

Comment:
These get set even when only the graph is shown so that the legend nows what data has been requested.